celery/worker/consumer/consumer.py

```diff
@@ -323,12 +324,21 @@ def start(self):
             try:
                 blueprint.start(self)
             except self.connection_errors as exc:
-                # If we're not retrying connections, no need to catch
-                # connection errors
-                if not self.app.conf.broker_connection_retry:
-                    raise
+                # If we're not retrying connections, we need to properly shutdown or terminate
+                # the Celery main process instead of abruptly aborting the process without any cleanup.
+                is_connection_loss_on_startup = self.restart_count == 0
+                connection_retry_type = ('broker_connection_retry_on_startup'
+                                         if (is_connection_loss_on_startup
+                                             and self.app.conf.broker_connection_retry_on_startup is not None)
+                                         else 'broker_connection_retry')
+                connection_retry = self.app.conf[connection_retry_type]
+                if not connection_retry:
+                    raise WorkerShutdown(
+                        f"Retrying to {'establish' if is_connection_loss_on_startup else 're-establish'} "
+                        f"a connection to the message broker after a connection loss has "
+                        f"been disabled (app.conf.{connection_retry_type}=False). Shutting down...") from exc
```

Comment:
@naomielst You should extract the message to a critical log entry (Just call [crit](https://github.com/celery/celery/blob/c889a5cb7606aec57708247aa05934d0b2dab4fc/celery/worker/consumer/consumer.py#L42) with the message as an argument)

```suggestion
                    crit(f"Retrying to {'establish' if is_connection_loss_on_startup else 're-establish'} "
                        f"a connection to the message broker after a connection loss has "
                        f"been disabled (app.conf.{connection_retry_type}=False). Shutting down...")
                    raise WorkerShutdown(1) from exc
```

The same goes for the other shutdown messages. Celery doesn't print these messages although it should since these exceptions should behave just like `sys.exit()` would.

Reply:
Done.